openpype/hosts/maya/api/lib.py

```diff
@@ -1791,6 +1799,24 @@ def _get_parent(node):
         if _id:
             return _id
 
+    if not history_only:
+        # Get siblings of same type
+        similar_nodes = cmds.listRelatives(parent,
+                                           type=node_type,
+                                           fullPath=True)
+        similar_nodes = cmds.ls(similar_nodes, exactType=node_type, long=True)
+
+        # Exclude itself
+        similar_nodes = [x for x in similar_nodes if x != node]
+
+        for similar_node in similar_nodes:
+            # Check if "intermediate object"
+            if cmds.getAttr(similar_node + ".intermediateObject"):
+                _id = get_id(similar_node)
+                if _id:
+                    return _id
```

Comment:
Should we log a warning if multiple intermediate shapes are found that have different ids with the info that we're taking only one of them? Just so we are aware of those conflicts? The other logic with the "history" took the first one (which is the closest one in history) which makes sense to prioritize. But for siblings I'm not so sure how we could correctly decide which one should be more likely to be correct?

Reply:
Indeed, so I think this is probably the best solution:
Set the default value to ```False``` ("allow_history_only"), with a warning in the particular case of more than 1 matching sibling.
I will make the changes (add the warning) if you all agree with that.